hail/python/hail/expr/types.py

```diff
@@ -1450,7 +1450,9 @@ def is_container(t) -> bool:
 def is_compound(t) -> bool:
     return (is_container(t)
             or isinstance(t, tstruct)
-            or isinstance(t, ttuple))
+            or isinstance(t, ttuple)
+            or isinstance(t, tndarray))
```

Comment:
What does compound mean in this case? I think of tndarray as a container type, but perhaps I don't understand what a container type means.

Reply:
I placed it here because there is a `ContainerExpression` which is extended by `ArrayExpression` and `TupleExpression` that contains a lot of logical assumptions that don't make sense for NDArrays. Would probably be good to revisit the structure of `ContainerExpression` though.

Reply:
ok. But what is a compound type again?

Reply:
An expression composed of other expressions

Reply:
got it.